test/integration/kubernetes_deploy_test.rb

```diff
@@ -381,8 +418,27 @@ def test_output_when_switching_labels_incorrectly
       web["spec"]["template"]["metadata"]["labels"] = { "name" => "foobar" }
     end
     assert_equal false, success
-    assert_logs_match("one of your templates is invalid")
-    assert_logs_match(/The Deployment "web" is invalid.*`selector` does not match template `labels`/)
+
+    assert_logs_match_all([
+      "one of your templates is invalid",
+      /The Deployment "web" is invalid.*`selector` does not match template `labels`/
+    ])
+  end
+
+  def test_can_deploy_deployment_with_zero_replicas
```

Comment:
I was going to mention that it's extremely important to make sure that it works with multiple long-running revisions / pods that take more time to shutdown / custom terminationGracePeriod, but it's already tested by `test_long_running_deployment` and if the test passes I don't have much to add.